itest/rust/src/rid_test.rs

```diff
@@ -0,0 +1,129 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{collections::HashSet, thread};
+
+use godot::{
+    engine::{Engine, RenderingServer},
+    prelude::{inner::InnerRid, Color, Rid, Vector2},
+};
+
+use crate::itest;
+
+/// Disable printing errors from Godot. Ideally we should catch and handle errors, ensuring they happen.
+/// But that isn't possible, so for now we just disable printing the error to avoid spamming the terminal.
+fn should_error(mut f: impl FnMut()) {
+    Engine::singleton().set_print_error_messages(false);
+    f();
+    Engine::singleton().set_print_error_messages(true);
+}
```

Comment:
Given the semantics, this could be named `suppress_godot_print` rather than `should_error`. 
Would it make sense to move it next to `expect_panic`?